tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -2174,15 +2174,19 @@ void IrEmitterUnnested::EmitPrologueForReduction(
     llvm::Type* element_type =
         llvm_ir::PrimitiveTypeToIrType(reduce_inst->shape().element_type(),
                                        ir_emitter_context_->llvm_module());
-    llvm::AllocaInst* reduction_input_address = Alloca(element_type);
+    llvm::AllocaInst* reduction_input_address =
+        llvm_ir::EmitAllocaAtFunctionEntry(element_type, absl::string_view(),
+                                           &b_);
     reduction_input_addresses->push_back(reduction_input_address);
 
     int num_partial_results = reduction_info->GetNumPartialResults();
     AddressVector* partial_result_addresses =
         reduction_info->GetMutablePartialResultAddresses();
+    std::string alloca_name = "partial_reduction_result." + std::to_string(i);
     llvm::AllocaInst* partial_result_address =
-        Alloca(element_type, /*ArraySize=*/b_.getInt32(num_partial_results),
-               "partial_reduction_result." + llvm::Twine(i));
+        llvm_ir::EmitAllocaAtFunctionEntryWithCount(
+            element_type, /*ArraySize=*/b_.getInt32(num_partial_results),
+            absl::string_view(alloca_name), &b_);
```

Comment:
`absl::string_view` should be able to take `std::string` and implicitly convert it.

Reply:
true...but was worried about whether about the lifetime of the underlying string object if I did a `absl::string_view("partial_reduction_result." + std::to_string(i))`

i.e. if the temporary string object ( `"partial_reduction_result." + std::to_string(i)` ) gets blown away before the call to `llvm_ir::EmitAllcoaAtFunctionEntry` completes, the the `absl::string_view` can be left pointing to a deleted object

Reply:
the life-cycle should be managed by `llvm::Twine`. and use `Twine.str()` to finish the concatenation and pass the copied result to `string_view`.

Reply:
can we remove `std::string` above, and directly pass `"partial_reduction_result." + llvm::Twine(i)).str()`

Reply:
or pass `alloca_name.str()` where `alloca_name` is of type `llvm::Twine`.

Reply:
I had this fixed...but forgot to commit before doing this push!